lib/rules/object-curly-newline.js

```diff
@@ -125,32 +140,59 @@ module.exports = {
 
         /**
          * Reports a given node if it violated this rule.
-         *
-         * @param {ASTNode} node - A node to check. This is an ObjectExpression node or an ObjectPattern node.
-         * @param {{multiline: boolean, minProperties: number}} options - An option object.
+         * @param {ASTNode} node - A node to check. This is an ObjectExpression, ObjectPattern, ImportDeclaration or ExportNamedDeclaration node.
+         * @param {{multiline: boolean, minProperties: number, consistent: boolean}} options - An option object.
          * @returns {void}
          */
         function check(node) {
             const options = normalizedOptions[node.type];
+
+            if (
+                (node.type === "ImportDeclaration" &&
+                    !node.specifiers.some(specifier => specifier.type === "ImportSpecifier")) ||
+                (node.type === "ExportNamedDeclaration" &&
+                    !node.specifiers.some(specifier => specifier.type === "ExportSpecifier"))
+            ) {
+                return;
+            }
+
             const openBrace = sourceCode.getFirstToken(node, token => token.value === "{");
+
             let closeBrace;
 
             if (node.typeAnnotation) {
                 closeBrace = sourceCode.getTokenBefore(node.typeAnnotation);
-            } else {
+            } else if (node.type === "ObjectExpression" || node.type === "ObjectPattern") {
                 closeBrace = sourceCode.getLastToken(node);
+            } else {
+                closeBrace = sourceCode.getTokenAfter(node.specifiers.slice(-1)[0], token => token.value === "}");
```

Comment:
Could we use `sourceCode.getLastToken(node, token => token.value === "}")` in both this code path and the one for ObjectExpression/ObjectPattern?